### DISCUSSION TITLE
Make use of `libmacchina`

### DISCUSSION THREAD

**Discussion by Gobidev** [LivacoNew/CrabFetch#27] [General]
I just wanted to let you know that there is a crate that implements many of the readouts you are fetching, called [`libmacchina`](https://crates.io/crates/libmacchina) ([github link](https://github.com/Macchina-CLI/libmacchina/)).

It probably makes sense to implement some metrics on your own, e.g. if you want them in a different format, but it is probably useful for implementing things like package count, as it already has support for many package managers with a strong focus on speed.

**Comment by LivacoNew** [comment]
While I'll likely keep it as a nice reference point for figuring out some detections, I won't be adding it as a used dep for a few reasons;

- While there is a performance gain of about 500µs from my light benchmarking, I would say this is due to macchina only grabbing the _counts_ of packages whereas CrabFetch grabs a full list of every package including names/versions. 
Crab does this so version detection has a speed-up, allowing it to simply use the already queried data from the package managers if the binary it's versioning is in `/usr/bin` which is a much more valuable time-save than that 500µs macchina has.
- As well as this, I can't see that much of a use for this anyways, most of it's readouts that I've spotted are already very easy to detect (memory, cpu, kernel, etc). No real point to using it for these simple detections that are only a few lines of logic to get anyways.
- Oh, and finally it's much less fun to use a library than to figure it out myself ;)
